### PR TITLE
Implement manufacturer_string and product_string

### PR DESCRIPTION
**Pull request by Antti**
Also fix serial_number.

This uses standard `wcstombs` function to convert between wide chars to multibyte chars.

### CODE REVIEW

File: src/device.rs [hunk 1]
```diff
@@ -3,9 +3,15 @@ use std::path::Path;
 use std::ffi::CStr;
 
 use sys::*;
+use libc;
 use Result as Res;
 use {Error, Handle};
 
+
+extern {
+	fn wcstombs(buf: *mut i8, pwcs: *const libc::wchar_t, n: libc::size_t) -> libc::size_t;
```
Comment (meh) [inline src/device.rs]: Isn't this available in the `libc` crate?

Comment (Antti) [reply src/device.rs]: Unfortunately no. I would be happy to use it from there.

**Comment by meh** [comment]
You should send a pull request to libc, they usually merge them pretty fast, I'd rather not define missing externs from libc in the crate 🐼

**Comment by Antti** [comment]
Makes sense. I will open PR there, and then update this PR to use `wcstombs` from `libc`:

https://github.com/rust-lang/libc/pull/510

**Comment by Antti** [comment]
@meh `libc` just bumped version. Now using wcstombs from libc.

**Comment by meh** [comment]
I don't think it's wortwhile to return a `Result` there, the conversion should never fail there, if it does there's something very wrong with the driver.

**Comment by meh** [comment]
I merged this but there seems to be an issue, apparently `serial_number` can be `NULL`, right now I made all of them return an `Option<String>` but I have doubts.

Do you know if `manufacturare_string` and `product_string` can be `NULL` as well?

**Comment by meh** [comment]
Disregard that, read the sources and it's clear they're optional.

**Pull request closed without merging**